bokeh/util/compiling.py

```diff
@@ -0,0 +1,274 @@
+from __future__ import absolute_import
+
+import logging
+logger = logging.getLogger(__name__)
+
+import six
+import json
+import inspect
+import hashlib
+from os.path import basename, dirname, join, abspath, relpath, exists
+from subprocess import Popen, PIPE
+
+from ..model import Model
+from ..settings import settings
+from .string import snakify
+from .paths import bokehjsdir
+
+# XXX: this is the same as bokehjs/src/js/plugin-prelude.js
+_plugin_prelude = \
+"""
+(function outer(modules, cache, entry) {
+  if (typeof Bokeh !== "undefined") {
+    for (var name in modules) {
+      Bokeh.require.modules[name] = modules[name];
+    }
+
+    for (var i = 0; i < entry.length; i++) {
+        Bokeh.Models.register_locations(Bokeh.require(entry[i]));
+    }
+  } else {
+    throw new Error("Cannot find Bokeh. You have to load it prior to loading plugins.");
+  }
+})
+"""
```

Comment:
Of course possible in at least a few ways, but for now wasn't significant enough to do. Also, the code was slowly converting from a different implementation to what plugins use. This outcome wasn't that obvious during early work.